Make Worksheet.append_table() return the response

ktpss95112 opened the pull request:
Hi nithinmurali, thanks for maintaining this project!
I opened issue #546 a few days ago. The issue is about returning the response of `Worksheet.append_table()` so that the user can obtain the index of the appended row.
I made this PR to provide the feature. If it is inappropriate, please let me know.
Thanks!

nithinmurali commented:
As discussed in the linked issue:

we don't return the underlying raw json in any api calls. We always wrap the data in sensible data-structure and return them. In this case i would suggest that it should return [GridRange](https://pygsheets.readthedocs.io/en/stable/address.html#pygsheets.GridRange) appropriately.

The result should look something similar to this

```
{
    'tableRange': <GridRange Sheet1!A1:D4>,
    'updates': {
        'updatedRange': <GridRange Sheet1!A1:D4>,
        'updatedCells': 6,
        'updatedColumns': 3,
        'updatedRows': 2
    }
}
```

ktpss95112 commented:
Thanks for your review. I updated the code and it should return something like this:

```
{   
    'tableRange': <GridRange result!A1:B12>, 
    'updates': {
        'updatedCells': 1,
        'updatedColumns': 1,
        'updatedRange': <GridRange result!A13:Z1009>,
        'updatedRows': 1
    }
}
```

Also, I found some spelling mistakes in `worksheet.py` and fixed them.
